internal/monitor/monitor.go

```diff
@@ -0,0 +1,108 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	nodesReadyMessage              = "All nodes are ready"
+	clusterVersionAvailableMessage = "ClusterVersion is available"
+)
+
+type ClusterInstallStatus struct {
+	Installed            bool
+	ClusterVersionStatus string
+	NodesStatus          string
+}
+
+func (status *ClusterInstallStatus) String() string {
+	installStatus := "installing"
+	if status.Installed {
+		installStatus = "installed"
+	}
+	return fmt.Sprintf("Cluster is %s\nClusterVersion Status: %s\nNodes Status: %s", installStatus, status.ClusterVersionStatus, status.NodesStatus)
+}
+
+type GetInstallStatusFunc func(ctx context.Context, log logrus.FieldLogger, c client.Client) ClusterInstallStatus
+
+func GetClusterInstallStatus(ctx context.Context, log logrus.FieldLogger, c client.Client) ClusterInstallStatus {
+	cvAvailable, cvMessage, err := clusterVersionStatus(ctx, log, c)
+	if err != nil {
+		cvMessage = fmt.Sprintf("Failed to check cluster version status: %s", err)
+	}
+
+	nodesReady, nodesMessage, err := nodesStatus(ctx, log, c)
+	if err != nil {
+		nodesMessage = fmt.Sprintf("Failed to check node status: %s", err)
+	}
+
+	return ClusterInstallStatus{
+		Installed:            cvAvailable && nodesReady,
+		ClusterVersionStatus: cvMessage,
+		NodesStatus:          nodesMessage,
+	}
+}
+
+func clusterVersionStatus(ctx context.Context, log logrus.FieldLogger, c client.Client) (bool, string, error) {
```

Comment:
I wonder if this is a right way to go.
There is some point where cluster version didn't update its status yet i think but node will be ready already.
Maybe we should use some LCA CR to actually manage IBI installation? 
What i mean is that cluster version can even not to start yet when node is already ready, actually i think it will not start before node is ready. 
I think it has a pretty big chance to set installed before cluster actually starts all the pods.

Reply:
Yeah, I wanted to ask you about this because I'm still seeing a lot of cluster operators not ready when these conditions are fulfilled.

Is there something better I can check or just go with this for now (it's better than it was for sure)